File: 14-LongestCommonPrefix/14-LongestCommonPrefix.py
# Last updated: 4/29/2026, 10:36:49 AM
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        if not strs or len(strs) == 0:
            return ""
        res = []
        len_list = list(map(lambda s:len(s), strs))
        print(len_list)
        min_len = min(len_list)
        print(min_len)
        if min_len == 0:
            return ""
        is_break = False
        idx = 0
        while not is_break and idx < min_len:
            ch = None
            for str in strs:
                if ch == None:
                    ch = str[idx]
                else:
                    if ch != str[idx]:
                        is_break = True
                        break
            if not is_break:
                res.append(ch)
                idx += 1
        print(res)
        return "".join(res)

File: 14-LongestCommonPrefix/14-LongestCommonPrefix.js
// Last updated: 4/29/2026, 10:36:44 AM
/**
 * @param {string[]} strs
 * @return {string}
 */
var longestCommonPrefix = function(strs) {
    const minLen = Math.min(...strs.map(str => str.length));
    console.log(minLen);
    let prefix = '';
    for(let i = 0; i < minLen; i++) {
        const ch = strs[0][i];
        console.log(`ch=${ch}`)
        let res = true;
        for(let j = 1; j < strs.length; j++) {
            const str = strs[j];
            if(ch !== str[i]) {
                res = false;
                break;
            }
        }
        console.log(`res=${res}`)
        if(res) {
            prefix += ch;
        } else {
            break;    
        }
    }
    return prefix;
};